org.lflang/src/org/lflang/ast/FormattingUtils.java

```diff
@@ -60,25 +77,25 @@ public static String render(EObject object, int lineLength) {
             INDENTATION,
             singleLineCommentPrefix
         );
-        var optimizedRendering = ms.render(INDENTATION, singleLineCommentPrefix);
+        var optimizedRendering = ms.render(INDENTATION, singleLineCommentPrefix, codeMapTags, null);
         List<String> comments = optimizedRendering.unplacedComments().toList();
         return comments.stream().allMatch(String::isBlank) ? optimizedRendering.rendering()
             : lineWrapComments(comments, lineLength, singleLineCommentPrefix)
                 + "\n" + optimizedRendering.rendering();
     }
 
     /**
-     * Return the prefix that the formatter should use to mark the start of a
-     * single-line comment.
+     * Infer the target language of the object. It is fine for this to be wrong
+     * as long as it is right about whether the target is Python.
      */
-    private static String getSingleLineCommentPrefix(EObject object) {
+    private static Target inferTarget(EObject object) {
         if (object instanceof Model model) {
             var targetDecl = ASTUtils.targetDecl(model);
-            if (targetDecl != null && targetDecl.getName().toUpperCase().contains("PYTHON")) {
-                return "#";
+            if (targetDecl != null) {
+                return Target.fromDecl(targetDecl);
             }
         }
-        return "//";
+        return Target.C;
```

Comment:
Why does this return the C target as default? This seems wrong to me

Reply:
Should it return `null`?

Reply:
It is probably because sometimes (not in actual code generation but maybe in debugging), we try to format a snippet of the AST that is not a full `Model`. Therefore, the target declaration is unavailable, and so the target is unknown.

Reply:
Right. For that reason, I changed it so that a `RuntimeException` is thrown. Is this the right type of exception, or should it be something like an `IllegalArgumentException`?

Reply:
I would be fine with either, although an `IllegalArgumentException` would be more specific I suppose.

I suppose that if we really want to do it the "right" way, we would replace references to `EObject` with `Model` so that the object passed to the `render` function has to be a `Model`. Then the runtime exception would not be thrown. I do not know how much work it would be to make it that way however, and I would be fine deferring that refactoring until the current design actually becomes a problem.